src/overmap.cpp

```diff
@@ -4337,28 +4338,19 @@ void overmap::place_radios()
         for( int j = 0; j < OMAPY; j++ ) {
             tripoint_om_omt pos_omt( i, j, 0 );
             point_om_sm pos_sm = project_to<coords::sm>( pos_omt.xy() );
-
-            if( ter( pos_omt ) == "radio_tower" ) {
-                int choice = rng( 0, 2 );
-                switch( choice ) {
-                    case 0:
-                        message = string_format( _( "This is emergency broadcast station %d%d."
-                                                    "  Please proceed quickly and calmly to your designated evacuation point." ), i, j );
-                        radios.push_back( radio_tower( pos_sm, strength(), message ) );
-                        break;
-                    case 1:
-                        radios.push_back( radio_tower( pos_sm, strength(),
-                                                       _( "Head West.  All survivors, head West.  Help is waiting." ) ) );
-                        break;
-                    case 2:
-                        radios.push_back( radio_tower( pos_sm, strength(), "", radio_type::WEATHER_RADIO ) );
-                        break;
+            // Since location have id such as "radio_tower_1_north", we must check the beginning of the id
+            if( is_ot_match( "radio_tower", ter( pos_omt ), ot_match_type::prefix ) ) {
```

Comment:
Maybe we can use a flag here instead of checking a hardcoded id.

Reply:
I would like to, but the flags would be stored in ``overmap_special`` items, while in the overmap::place_radios I have only access to ``oter_id`` items. Is there any way to get the overmap_special from an oter_id?

Reply:
`oter_id` points to `struct oter_t` which in turn points to `struct oter_type_t`, which is `overmap_terrain` in json. Maybe you can add the flag to that instead (the id of `overmap_terrain` is the id without suffix it seems, for example `radio_tower_1`)

Reply:
I don't know, the flags of ``struct oter_type_t`` are stored as a bitset, rather than as a set of strings like most other objects in the game. If I want to add a flag, it means I must add it both in ``enum class oter_flags`` and in ``static const std::map<std::string, oter_flags> oter_flags_map``. Would it be alright?

Reply:
Well, that bitset actually having a mapping of flag strings to values is a sign of extendability to me. We can of course leave it for a later PR if things are too uncertain.

Reply:
I would prefer to have a confirmation. Thank for your feedback anyway!